InnovativeCars/force-app/main/default/classes/CarToShowroomHandler1.cls

```diff
@@ -16,26 +16,35 @@ public class CarToShowroomHandler1 {
     */
     public static void checkIfCarIsAlreadyAvailable(List<Car_To_Showroom__c> newOffers) {
         Set<Id> carIds = new Set<Id>();
+        Set<Id> showroomIds = new Set<Id>();
         for(Car_To_Showroom__c newOffer : newOffers) {
-            if(newOffer.Availability__c == 'Available at the showroom') {
+            if(newOffer.Availability__c == Constants.picklistAvailable) {
                 carIds.add(newOffer.Car__c);
+                showroomIds.add(newOffer.Showroom__c);
             }
         }
-        List<Car_To_Showroom__c> carToShowroomsList = new List <Car_To_Showroom__c>(
-            [SELECT Car__c, Showroom__c FROM Car_To_Showroom__c 
-             WHERE Availability__c LIKE 'Available at the showroom' 
-             AND Car__c IN :carIds]);
+
+        Map<Id, Showroom__c> showroomMap = new Map<Id, Showroom__c>([SELECT Id, Name FROM Showroom__c WHERE Id IN :showroomIds]);
+
+        List<Car_To_Showroom__c> carToShowroomsList = [SELECT Id, Car__c, Showroom__c FROM Car_To_Showroom__c 
```

Comment:
Moze warto sie zastanowic nad Showroom__r.Name wtedy select wyzej jest nie potrzebny a w mapie zamiast Car__c na id miala bys Car__c na Showroom__r.Name

Reply:
Czyli Map<Car__c, Showroom__r.Name>? Pytam, bo tak próbuję, jednak rzuca mi, że Showroom__r.Name jest invalid type

Reply:
Odwołuję pytanie, już wykombinowałam jak to zrobić :)